### DISCUSSION TITLE
controlled and uncontrolled problem

### DISCUSSION THREAD

**Discussion by TOcvfan** [react-hook-form/react-hook-form#3211] [Q&A]
My form is coming great but after I went from react datepicker to material-ui datepicker I first got a date format issue, but I maneged to solve it and it is actually working but right now it comes with this warning:
`A component is changing an uncontrolled input of type text to be controlled.`

I'm not sure how to solve this so any help you can give will be appreciated..
I have made a codesandbox here: [https://codesandbox.io/s/keen-poitras-qei27?file=/src/App.js](url)
Sorry about the language but it is supposed to be in a site were only danes come, so I naturally made it in danish :-)

**Comment by bluebill1049** [comment]
<img width="410" alt="Screen Shot 2020-10-20 at 10 31 54 am" src="https://user-images.githubusercontent.com/10513364/96522630-7a9b8c80-12bf-11eb-9a34-c4384c502a23.png">

I think you want to use `render` props

<img width="947" alt="Screen Shot 2020-10-20 at 10 32 30 am" src="https://user-images.githubusercontent.com/10513364/96522659-90a94d00-12bf-11eb-901f-c4c0da71edcd.png">

there is an example: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by TOcvfan** [reply]
Yes that fixed the controlled/uncontrolled problem but the reason that I tried it the other way is that when I submit I get an `Unhandled Rejection (RangeError): Invalid time value`  error..

**Reply by bluebill1049** [reply]
so the problem is probably around writing up with all the `props` from `render`
